module/documents/actor/actor.mjs

```diff
@@ -1779,18 +1779,30 @@ export default class Actor5e extends Actor {
    * carried by an Actor.
    * @returns {Promise<Actor5e>}
    */
+
   convertCurrency() {
     const curr = foundry.utils.deepClone(this.system.currency);
     const conversion = Object.entries(CONFIG.DND5E.currencies);
-    conversion.reverse();
-    for ( let [c, data] of conversion ) {
-      const t = data.conversion;
-      if ( !t ) continue;
-      let change = Math.floor(curr[c] / t.each);
-      curr[c] -= (change * t.each);
-      curr[t.into] += change;
-    }
-    return this.update({"system.currency": curr});
+    conversion.sort((a, b) => a[1].conversion - b[1].conversion);
+
+    var change = 0;
+    for ( let [c, data] of conversion) {
+      const rate = data.conversion;
+      if ( !rate ) continue;
+
+      change = change + curr[c] / rate;
+    }
```

Comment:
- No need to use `var`
- The configuration definition states that the `conversion` is always required, so we shouldn't need to verify it exists here
- This would probably be clearer as a reduce

```suggestion
    let change = conversion.reduce((change, [c, data]) => {
      return change + (currency[c] / data.conversion);
    }, 0);
```

Reply:
Thank you for these suggestions! They all look great, and I just have one comment on this one.

One use case I'm trying to support is the removal of currencies. Many players who use my module simply want to remove electrum and platinum. Until now, I had intended to support their need by allowing `conversion` to be undefined. That is why I explicitly check that `conversion` exists.

I'd like to get your advice on how best to support the need to remove currencies. 

One approach might be to advise 3rd-party modules to simply remove unneeded currency objects _entirely_ from `DND5E.currencies`. The currency converter code above would work as expected. However, any hardcoded references to specific currencies would break. Those hardcoded references would need to be changed to an approach that iterates through the currencies stored in `DND5E.currencies`.

Another approach might be to add a "removed" or "hidden" boolean value to the currency configuration schema. We would have to change the logic of the currency converter above to check for that value and skip any currencies that aren't supposed to exist.

Reply:
I think in order to function in the same way as removing `skills` or `abilities`, users should just need to remove a currency from the `CONFIG.currencies` object (or replace it entirely with their own set) and everything else should work.

There are a few changes I have planned with the upcoming data model stuff that should also help improve support for this. For now I'd say just worry about conversion stuff and we can take a look at anything else needed later.